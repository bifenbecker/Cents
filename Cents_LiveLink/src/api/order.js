import httpClient from "./httpClient";

export const fetchStoreSettings = params => {
  return httpClient({
    method: "GET",
    url: "/live-status/settings",
    params,
  });
};

export const authenticateOrder = token => {
  return httpClient({
    method: "GET",
    url: "/live-status/verify-order",
    params: {
      token,
    },
  });
};

export const fetchOrderDetail = token => {
  return httpClient({
    method: "GET",
    url: "/live-status",
    params: {
      token,
    },
  });
};

export const updateTip = (token, tipData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/tip",
    params: {
      token,
    },
    data: tipData,
  });
};

export const addPromo = (token, promoData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-promotion",
    params: {
      token,
    },
    data: promoData,
  });
};

export const removePromo = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-promotion",
    params: {
      token,
    },
  });
};

export const addCredits = (token, creditsData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-credits",
    params: {
      token,
    },
    data: creditsData,
  });
};

export const removeCredits = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-credits",
    params: {
      token,
    },
  });
};

export const setOnlineOrderReturnMethod = (token, data) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/return-method",
    params: {
      token,
    },
    data,
  });
};

export const updatePaymentMethodIntent = (token, {paymentToken}) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/payment-method",
    params: {
      token,
    },
    data: {paymentToken},
  });
};

export const cancelOrder = params => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/order/cancel",
    params,
  });
};

export const getRouteDelivery = (token, {orderDelivery}) => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-delivery/${orderDelivery}/route-details`,
    params: {token},
  });
};

export const fetchOrderDeliveries = token => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-deliveries`,
    params: {token},
  });
};

export const manageOrder = (token, data) => {
  return httpClient({
    method: "PUT",
    url: `/live-status/live-link/manage`,
    params: {token},
    data,
  });
};
