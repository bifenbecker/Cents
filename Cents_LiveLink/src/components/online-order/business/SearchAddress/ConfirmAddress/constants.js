export const PLACE_HOLDER = {
  ADDRESS_DETAILS: "Address details",
  PICKUP_INSTRUCTIONS:
    "e.g. call upon arrival, entrance on right side of building, don’t ring bell, etc.",
};

export const LABEL_TEXT = {
  ADDRESS_DETAILS: "Apt / Suite / Unit",
  PICKUP_INSTRUCTIONS: "Pickup Instructions",
};

export const SUBTITLE_TEXT = {
  ADDRESS_DETAILS: "Confirm your address for pickup",
  PICKUP_INSTRUCTIONS: "Pickup Instructions",
};

export const CHECKBOX_LABEL = "Leave it at my door if I’m not home";

export const GET_BTN_TEXT = (flag) => {
  return flag ? "Save & Continue" : "Save";
};

export const NOTES_TEXT = "We will send you updates about your pickup via text.";
