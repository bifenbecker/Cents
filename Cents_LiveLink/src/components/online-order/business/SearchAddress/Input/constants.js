export const SEARCH_FOR_A_NEW_ADDRESS = "Search for a new address";
export const SEARCH_FOR_AN_ADDRESS = "Search for an address";
export const SEARCH_ADDRESS = "Search Address";
