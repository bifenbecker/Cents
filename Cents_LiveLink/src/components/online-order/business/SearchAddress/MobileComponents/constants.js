export const STATES = {
  CONFIRM: "CONFIRM",
  CANT_PICKUP: "CANT_PICKUP",
  AUTO_COMPLETE: "AUTO_COMPLETE",
  SAVED_ADDRESSES: "SAVED_ADDRESSES",
};
