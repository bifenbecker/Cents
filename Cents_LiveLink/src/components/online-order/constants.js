export const orderPickUpSteps = {
  details: "order-details",
  finishingUp: "finishing-up",
};

export const deliveryProviders = {
  uber: "UBER",
  ownDriver: "OWN_DRIVER",
  doorDash: "DOORDASH",
};

export const orderFields = [
  "servicePriceId",
  "serviceModifierIds",
  "customerNotes",
  "isHangDrySelected",
  "hangDryInstructions",
  "orderNotes",
  "customerAddressId",
  "paymentToken",
  "promoCode",
  "bagCount",
  "returnMethod",
  "hasDryCleaning",
];

export const mandatoryOrderFields = [
  "paymentToken",
  "returnMethod",
  "servicePriceId",
  "customerAddressId",
];

export const bufferRequiredForOrder = {
  [deliveryProviders.uber]: 10,
  [deliveryProviders.doorDash]: 10,
  [deliveryProviders.ownDriver]: 20,
};

export const orderDeliveryFields = ["pickup", "delivery"];

export const orderDeliverySubFields = [
  "type",
  "timingsId",
  "deliveryProvider",
  "deliveryWindow",
  "thirdPartyDeliveryId",
  "totalDeliveryCost",
  "courierTip",
  "subsidyInCents",
  "thirdPartyDeliveryCostInCents",
];

export const onDemandDeliveryTypes = {
  pickup: "PICKUP",
  delivery: "DELIVERY",
  pickupAndDelivery: "PICKUP_AND_DELIVERY",
};

export const SERVICE_CATEGORY_TYPES = {
  DRY_CLEANING: "DRY_CLEANING",
  LAUNDRY: "LAUNDRY",
  ALTERATIONS: "ALTERATIONS",
};

export const cloneOrderState = {
  REVIEW: "REVIEW_CLONE_ORDER",
  EDIT: "EDIT_CLONE_ORDER",
  INITIAL: "NO_CLONE_ORDER",
};
