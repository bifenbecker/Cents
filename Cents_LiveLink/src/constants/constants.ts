export const OTP_CODE_LENGTH = 6;
export const PHONE_LENGTH = 10;

export const PHONE_NUMBER_FORMAT_REGEXP = /^(\d{3})(\d{3})(\d{4})$/;
export const NOT_DIGITS_REGEXP = /\D/g;
