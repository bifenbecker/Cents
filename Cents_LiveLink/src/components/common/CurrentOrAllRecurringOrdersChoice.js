import React, {useState, useMemo} from "react";
import PropTypes from "prop-types";
import {Flex, Text, Button} from "rebass/styled-components";

import useWindowSize from "../../hooks/useWindowSize";
import {orderChoices, orderChoicesDisplay} from "../../constants/order";

import {DockModal} from "./.";

const CurrentOrAllRecurringOrdersChoice = props => {
  const [, height] = useWindowSize();
  const {isOpen, toggle, dockProps, header, onSubmit} = props;

  const [selectedChoice, setSelectedChoice] = useState(orderChoices.currentOrder);

  const getSize = useMemo(() => {
    if (height >= 568) {
      return 364;
    } else {
      return 0.75 * height;
    }
  }, [height]);

  const closeModal = () => {
    setSelectedChoice(orderChoices.currentOrder);
    toggle();
  };

  const selectChoice = e => {
    setSelectedChoice(e.target.id);
  };

  const handleSubmitClick = () => {
    onSubmit(selectedChoice);
  };

  return (
    <DockModal
      {...dockProps}
      header={header}
      isOpen={isOpen}
      toggle={closeModal}
      size={getSize}
      fixedSize
      showExitIcon
      zIndex={10}
    >
      <>
        <Flex onClick={selectChoice} {...styles.wrapper}>
          {Object.values(orderChoices).map(choice => (
            <Text
              id={choice}
              sx={{
                ...styles.text,
                bg: selectedChoice === choice ? "BACKGROUND_LIGHT_BLUE" : "",
              }}
              key={choice}
            >
              {orderChoicesDisplay[choice]}
            </Text>
          ))}
        </Flex>
        <Flex {...styles.saveButtonContainer}>
          <Button {...styles.saveButton} onClick={handleSubmitClick}>
            Ok
          </Button>
        </Flex>
      </>
    </DockModal>
  );
};

const styles = {
  wrapper: {
    flexDirection: "column",
    justifyContent: "space-between",
  },
  text: {
    paddingLeft: "37px",
    py: "33px",
  },
  saveButtonContainer: {
    sx: {
      margin: "0",
      minWidth: "0",
      position: "relative",
      width: "100%",
      bottom: 0,
      left: 0,
      marginTop: "auto",
      boxShadow: "0 -5px 8px -7px rgba(0,0,0,0.2)",
      bg: "WHITE",
    },
  },
  saveButton: {
    sx: {
      backgroundColor: "#3D98FF",
      width: "100%",
      borderRadius: 31,
      textTransform: "uppercase",
      marginLeft: "18px",
      marginRight: "18px",
      boxShadow: "0px 5px 25px rgba(121, 120, 120, 0.248907)",
    },
    my: 34,
    py: 20,
  },
};

CurrentOrAllRecurringOrdersChoice.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  dockProps: PropTypes.object,
  header: PropTypes.string.isRequired,
};

CurrentOrAllRecurringOrdersChoice.defaultProps = {
  dockProps: {},
};

export default CurrentOrAllRecurringOrdersChoice;
