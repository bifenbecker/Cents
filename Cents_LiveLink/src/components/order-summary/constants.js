import {RETURN_METHODS} from "../../constants/order";
import {
  CALL_ICON,
  DELIVERY_ORDER_COMPLETED,
  ON_DEMAND_DELIVERY_IN_PROGRESS,
  ON_DEMAND_PICKUP_COMPLETED,
  ON_DEMAND_PICKUP_INITIATED,
  ON_DEMAND_PICKUP_SCHEDULED,
  ORDER_CANCELED,
  ORDER_COMPLETED,
  PROCESSING,
  READY_FOR_CUSTOMER_PICKUP,
  SMS_ICON,
  STD_DELIVERY_IN_PROGRESS,
  STD_PICKUP_COMPLETED,
  STD_PICKUP_INITIATED,
  STD_PICKUP_SCHEDULED,
} from "../../assets/images";

export {RETURN_METHODS, ORDER_TYPES} from "../../constants/order";

export const RESIDENTIAL_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Received & Processing",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const ONLINE_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Received & Processing",
  readyForPickup: "Laundry Ready",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const SERVICE_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Processing",
  readyForPickup: "Laundry Ready",
  completed: "Complete",
};

export const DELIVERY_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Processing",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const TIMELINE_CANCELED_STATUS = "Canceled";

export const DELIVERY_TRACKING_ORDER_STATUSES = {
  submitted: "SUBMITTED",
  enRouteToPickup: "EN_ROUTE_TO_PICKUP",
  enRouteToDropoff: "EN_ROUTE_TO_DROPFF",
  completed: "COMPLETED",
};

export const RETURN_METHODS_DISPLAY = {
  [RETURN_METHODS.inStorePickup]: "NO THANKS",
  [RETURN_METHODS.delivery]: "DELIVER TO ME",
};

export const serviceSelections = {
  laundry: "Laundry",
  dryCleaning: "Dry Cleaning",
};

export const orderItemCategories = {
  deliveryOrPickup: "DELIVERY",
  fixedPriceItem: "FIXED_PRICE",
  perPoundItem: "PER_POUND",
};

export const ORDER_STATUSES = {
  COMPLETED: "COMPLETED",
  CANCELLED: "CANCELLED",
};

export const IMAGE_IDS = {
  DELIVERY_ORDER_COMPLETED: DELIVERY_ORDER_COMPLETED,
  ON_DEMAND_DELIVERY_IN_PROGRESS: ON_DEMAND_DELIVERY_IN_PROGRESS,
  ON_DEMAND_PICKUP_COMPLETED: ON_DEMAND_PICKUP_COMPLETED,
  ON_DEMAND_PICKUP_INITIATED: ON_DEMAND_PICKUP_INITIATED,
  ON_DEMAND_PICKUP_SCHEDULED: ON_DEMAND_PICKUP_SCHEDULED,
  ORDER_CANCELED: ORDER_CANCELED,
  ORDER_COMPLETED: ORDER_COMPLETED,
  PROCESSING: PROCESSING,
  READY_FOR_CUSTOMER_PICKUP: READY_FOR_CUSTOMER_PICKUP,
  STD_DELIVERY_IN_PROGRESS: STD_DELIVERY_IN_PROGRESS,
  STD_PICKUP_COMPLETED: STD_PICKUP_COMPLETED,
  STD_PICKUP_SCHEDULED: STD_PICKUP_SCHEDULED,
  STD_PICKUP_INITIATED: STD_PICKUP_INITIATED,
  CALL_ICON,
  SMS_ICON,
};

export const PAYMENT_STATUSES = {
  balanceDue: "BALANCE_DUE",
  paid: "PAID",
  pending: "PENDING",
};
