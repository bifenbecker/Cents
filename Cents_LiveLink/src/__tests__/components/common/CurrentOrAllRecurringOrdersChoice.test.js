import React from "react";
import {fireEvent, render} from "@testing-library/react";

import CurrentOrAllRecurringOrdersChoice from "../../../components/common/CurrentOrAllRecurringOrdersChoice";
import {orderChoicesDisplay, orderChoices} from "../../../constants/order";

const onSubmit = jest.fn();
const toggle = jest.fn();

const renderCurrentOrAllRecurringOrdersChoice = () => {
  const {getByText, container, getByAltText} = render(
    <CurrentOrAllRecurringOrdersChoice
      {...{
        isOpen: true,
        toggle,
        header: "Some Header",
        onSubmit,
      }}
    />
  );

  return {getByText, getByAltText, container};
};

describe("CurrentOrAllRecurringOrdersChoice Component", () => {
  it("should match snaphot", () => {
    const {container} = renderCurrentOrAllRecurringOrdersChoice();

    expect(container).toMatchSnapshot();
  });

  it("should render header", () => {
    const {getByText} = renderCurrentOrAllRecurringOrdersChoice();

    expect(getByText("Some Header")).toBeInTheDocument();
  });

  it("should trigger toggle when close icon is clicked", () => {
    const {getByAltText} = renderCurrentOrAllRecurringOrdersChoice();

    const closeIcon = getByAltText("Dock Close");
    expect(closeIcon).toBeInTheDocument();

    fireEvent.click(closeIcon);

    expect(toggle).toBeCalled();
  });

  it("should call onSubmit with currentAndFutureOrders as selected choice", () => {
    const {getByText} = renderCurrentOrAllRecurringOrdersChoice();
    fireEvent.click(getByText(orderChoicesDisplay.currentAndFutureOrders));
    const button = getByText("Ok");
    fireEvent.click(button);

    expect(onSubmit).toHaveBeenCalled();
    expect(onSubmit).toHaveBeenCalledWith(orderChoices.currentAndFutureOrders);
  });

  it("should call onSubmit with currentOrder as selected choice", () => {
    const {getByText} = renderCurrentOrAllRecurringOrdersChoice();
    fireEvent.click(getByText(orderChoicesDisplay.currentOrder));
    const button = getByText("Ok");
    fireEvent.click(button);

    expect(onSubmit).toHaveBeenCalled();
    expect(onSubmit).toHaveBeenCalledWith(orderChoices.currentOrder);
  });
});
