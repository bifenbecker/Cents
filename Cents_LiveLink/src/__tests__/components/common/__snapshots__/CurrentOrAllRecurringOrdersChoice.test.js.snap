// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentOrAllRecurringOrdersChoice Component should match snaphot 1`] = `
<div>
  <div
    style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
  >
    <div
      style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
    />
    <div
      style="position: fixed; z-index: 10; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
    >
      <div
        style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
      />
      <div
        style="width: 100%; height: 100%; overflow: auto;"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi haujOk hzPtvw"
          font-size="18px"
          height="67px"
        >
          <img
            alt="Dock Close"
            class="sc-bczRLJ FYQvo"
            src="Icon_Mobile_Exit_Flow.svg"
          />
          <div
            class="sc-bczRLJ flhfNA"
            color="black"
            font-size="18px"
          >
            Some Header
          </div>
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi kXDPzY hzPtvw"
        >
          <div
            class="sc-bczRLJ clirAX"
            id="currentOrder"
          >
            This order only
          </div>
          <div
            class="sc-bczRLJ kbeXWO"
            id="currentAndFutureOrders"
          >
            This and future recurring orders
          </div>
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi kUrYHc hzPtvw"
        >
          <button
            class="sc-bczRLJ dxgfIt"
          >
            Ok
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
